packages/dart/pubspec.yaml

```diff
@@ -24,6 +24,7 @@ dependencies:
   mime_type: ^1.0.0
 
 dev_dependencies:
+  lints: ^1.0.1
```

Comment:
Should we remove the range operators for consistency?

Reply:
In order to make sure the tests always run using the same versions?
We could do that, but for the dependencies I would keep them, since dependency hell would strike soon.

Reply:
It would be best to keep it fixed and use an external service to upgrade it on demand, like Snyk does for npm dependencies. Unfortunately it doesn't support dart dependencies. Do you know of any 3rd party service for dart?

Reply:
Do you want to remove the range operators just for the dev_dependencies or for all dependencies?
I would not remove them for all, since it gives `pub get` more options for finding versions that meet all requirements.

Regarding a tool, I don't know one, that supports it, but I already thought about adding such a functionality to [uptodate](https://github.com/fischerscode/uptodate).

Reply:
Could you explain how the lock file works here, is that updated on every publish or is it a distinct command that we should run from time to time to keep the dependencies updated at least within the range that is specified in `pubspec.yaml`?

Reply:
I guess this is a separate discussion unrelated to this PR, so I'll resolve this for now.

Reply:
TLDR;
Pub checks what are the best versions it can use.

Pub is the package manager of dart/flutter. "pubspec.yaml" sets the the dependencies the project needs. The dependencies are the ones the code uses. dev_dependencies are used for tests and stuff during development.
Packages, like this project, usually don't version control the 'pubspec.lock' file. Applications usually do.
When `pub get` runs, it determines the latest versions for all the (nested) dependencies. Since the versions are usually specified like `package: ^1.2.3`, which means every stable version greater or equal to 1.2.3 but smaller then 2.0.0, pub has some room to update the dependencies without running into breaking changes. This is needed, since packages might share dependencies and not every package is 100% up to date.
The versions used get stored in `pubspec.lock`. The next time `pub get` runs, it just checks if `pubspec.lock` still meets the requirements of `pubspec.yaml`. If not, versions are determined again.
`pub upgrade` can be used to force updates to `pubspec.lock`.
`pub outdated` can list possible updates (including major changes).

Additionally the developer has the option to force override a dependency, even if it does not satisfy the requirements of other (nested) dependencies.

Reply:
> The next time pub get runs, it just checks if pubspec.lock still meets the requirements of pubspec.yaml. If not, versions are determined again.

Let's assume this scenario:

- dependency version in pubspec.yaml: ^1.0.0
- dependency version in pubspec.lock: 1.0.1
- latest dependency version available: 1.0.2

Then I publish a release and the workflow runs `pub get`, what happens? Is pubspec.lock updated to 1.0.2 or does it stay 1.0.1?